---
title: "Phylogenetic Diversity - Traits"
author: 'Ashwini Ramesh; Z620: Quantitative Biodiversity, Indiana University'
geometry: margin=2.54cm
---

## OVERVIEW

Up to this point, we have been focusing on patterns taxonomic diversity in Quantitative Biodiversity. 
Although taxonomic diversity is an important dimension of biodiversity, it is often necessary to consider the evolutionary history or relatedness of species. 
The goal of this exercise is to introduce basic concepts of phylogenetic diversity. 

After completing this exercise you will be able to:

1.  create phylogenetic trees to view evolutionary relationships from sequence data
2.  map functional traits onto phylogenetic trees to visualize the distribution of traits with respect to evolutionary history
3.  test for phylogenetic signal within trait distributions and trait-based patterns of biodiversity

## Directions:

1. Change "Student Name" on line 3 (above) with your name.
2. Complete as much of the exercise as possible during class; what you do not complete in class will need to be done on your own outside of class.
3. Use the handout as a guide; it contains a more complete description of data sets along with the proper scripting needed to carry out the exercise.
4. Be sure to **answer the questions** in this exercise document; they also correspond to the handout.
Space for your answer is provided in this document and indicated by the ">" character.
If you need a second paragraph be sure to start the first line with ">".
5. Before you leave the classroom, **push** this file to your GitHub repo.
6. For homework, follow the directions at the bottom of this file. 
7. When you are done, **Knit** the text and code into a PDF file.
8. After Knitting, please submit the completed exercise by creating a **pull request** via GitHub.
Your pull request should include this file *PhyloTraits_exercise.Rmd* and the PDF output of `Knitr` (*PhyloTraits_exercise.pdf*).


## 1) SETUP

Typically, the first thing you will do in either an R script or an RMarkdown file is setup your environment. 
This includes things such as setting the working directory and loading any packages that you will need.

In the R code chunk below, provide the code to:  
1. clear your R environment,  
2. print your current working directory,  
3. set your working directory to your "*/Week6-PhyloTraits*" folder, and  
4. load all of the required R packages (be sure to install if needed).  

```{r}
rm(list=ls())
getwd()
#setwd("~ /Users/DELL/GitHub/QB2017_Ramesh/Week6-PhyloTraits")

package.list <- c("ape","seqinr","phylobase","adephylo","geiger","picante","stats","RColorBrewer","caper","phylolm","pmc","ggplot2","tidyr","dplyr","phangorn","pander")
for(package in package.list){
  if(!require(package,character.only = TRUE, quietly = TRUE)){
    install.packages(package,repos = "http://cran.us.r-project.org")
    library(package,character.only = TRUE)
  }
  
}
```

## 2) DESCRIPTION OF DATA

The maintenance of biodiversity is thought to be influenced by **trade-offs** among species in certain functional traits. 
One such trade-off involves the ability of a highly specialized species to perform exceptionally well on a particular resource compared to the performance of a generalist. 
In this exercise, we will take a phylogenetic approach to mapping phosphorus resource use onto a phylogenetic tree while testing for specialist-generalist trade-offs. 


## 3) SEQUENCE ALIGNMENT

***Question 1***: Using less or your favorite text editor, compare the `p.isolates.fasta` file and the `p.isolates.afa` file. 
Describe the differences that you observe between the files. 

***Answer 1***: The `p.isolates.afa` file has all sequences aligns to one another; while the `p.isolates.fasta` file contains only the reads (not aligned)

In the R code chunk below, do the following:
1. read your alignment file,
2. convert the alignment to a DNAbin object,
3. select a region of the gene to visualize (try various regions), and
4. plot the alignment using a grid to visualize rows of sequences.

```{r}
# Read Alignmnet File
read.aln <- read.alignment(file = "./data/p.isolates.afa",format = "fasta")
p.DNAbin <- as.DNAbin(read.aln)
window <- p.DNAbin[,0:1000]
image.DNAbin(window, cex.lab = 0.5)
```

***Question 2***:  Make some observations about the `muscle` alignment of the 16S rRNA gene sequences for our bacterial isolates and the outgroup, *Methanosarcina*, a member of the domain archaea. 
Move along the alignment by changing the values in the `window` object. 

a. Approximately how long are our reads?  
b. What regions do you think would are appropriate for phylogenetic inference and why?  


> ***Answer 2a***:   ~ 700 base pairs

> ***Answer 2b***:  Regions with high number of variable regions and preferable with no loss of data are ideal for phylogenetic inferences as these are the regions with SNPs and are responsible for any variation in genetic/ functional make-up; leading to to speciation events. 
  

## 4) MAKING A PHYLOGENETIC TREE

Once you have aligned your sequences, the next step is to construct a phylogenetic tree.
Not only is a phylogenetic tree effective for visualizing the evolutionary relationship among taxa, but as you will see later, the information that goes into a phylogenetic tree is needed for downstream analysis. 

### A. Neighbor Joining Trees

In the R code chunk below, do the following:  
1. calculate the distance matrix using `model = "raw"`,  
2. create a Neighbor Joining tree based on these distances,  
3. define "Methanosarcina" as the outgroup and root the tree, and  
4. plot the rooted tree. 

```{r}
seq.dist.raw <- dist.dna(p.DNAbin, model  ="raw", pairwise.deletion = FALSE)

nj.tree <- bionj(seq.dist.raw)
outgroup <- match("Methanosarcina", nj.tree$tip.label)
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

par(mar =c(1,1,2,1) + 0.1)
plot.phylo(nj.rooted, main = "Neighbourhood Joining Treee", "phylogram", use.edge.length = FALSE,direction = "right",cex =0.6, label.offset = 1)
add.scale.bar(cex = 0.7)

```

***Question 3***: What are the advantages and disadvantages of making a neighbor joining tree?   


> ***Answer 3***:  Advantage: NJ trees are fast to simulate as they rely on the distance matrix between species. Disadvantage: It does not account for the type of nucleotides, not statistically robust, dependent on type of substitution model used to design it.  

### B) SUBSTITUTION MODELS OF DNA EVOLUTION 

In the R code chunk below, do the following:  
1. make a second distance matrix based on the Felsenstein 84 substitution model,  
2. create a saturation plot to compare the *raw* and *Felsenstein (F84)* substitution models,  
3. make Neighbor Joining trees for both, and  
4. create a cophylogenetic plot to compare the topologies of the trees.

```{r}
seq.dist.F84 <- dist.dna(p.DNAbin,model = "F84",pairwise.deletion = FALSE)
par(mar=c(5,5,2,1)+0.1)
plot(seq.dist.raw, seq.dist.F84,pch = 20, col = "red", las = 1, asp = 1,xlim = c(0,0.7), ylim = c(0,0.7), xlab = "Raw Distance", ylab = "F84 Distance")
abline(b=1,a=0,lty=2)
text(0.65,0.6,"1:1")
```

In the R code chunk below, do the following:  
1. pick another substitution model,  
2. create and distance matrix and tree for this model,  
3. make a saturation plot that compares that model to the *Felsenstein (F84)* model,  
4. make a cophylogenetic plot that compares the topologies of both models, and  
5. be sure to format, add appropriate labels, and customize each plot.

```{r}
seq.dist.K80 <- dist.dna(p.DNAbin,model = "K80",pairwise.deletion = FALSE)
par(mar=c(5,5,2,1)+0.1)
plot(seq.dist.K80, seq.dist.F84,pch = 20, col = "red", las = 1, asp = 1,xlim = c(0,0.7), ylim = c(0,0.7), xlab = "K20 Distance", ylab = "F84 Distance")
abline(b=1,a=0,lty=2)
text(0.65,0.6,"1:1")

K80.tree <- bionj(seq.dist.K80)
F84.tree <- bionj(seq.dist.F84)

K80.outgroup <- match("Methanosarcina", K80.tree$tip.label)
F84.outgroup <- match("Methanosarcina", F84.tree$tip.label)

K80.rooted <- root(K80.tree, K80.outgroup, resolve.root  = TRUE)
F84.rooted <- root(F84.tree, F84.outgroup, resolve.root  = TRUE)

layout(matrix(c(1,2),1,2),width = c(1,1))
par(mar=c(1,1,2,0))
plot.phylo(K80.rooted, type = "phylogram", direction = "right",show.node.label = TRUE,use.edge.length = FALSE, adj = 0.5, cex = 0.6, label.offset = 2,main = "K80" )

par(mar=c(1,0,2,1))
plot.phylo(F84.rooted, type = "phylogram", direction = "right",show.node.label = TRUE,use.edge.length = FALSE, adj = 0.5, cex = 0.6, label.offset = 2,main = "F84" )

dist.topo(K80.rooted, F84.rooted, method = "score")
```

***Question 4***:  

a. Describe the substitution model that you chose. What assumptions does it make and how does it compare to the F84 model?
b. Using the saturation plot and cophylogenetic plots from above, describe how your choice of substitution model affects your phylogenetic reconstruction. 
If the plots are inconsistent with one another, explain why.
c. How does your model compare to the *F84* model and what does this tell you about the substitution rates of nucleotide transitions?

	
> ***Answer 4a***:   K80 Model: All base frequencies are same; P(Transition mutation) > P(Transversion mutation)
F84 : Base frequencies, rate of Transition, rate of Transversion are different

> ***Answer 4b***:   Both the plots show that the two models are highly consistent with one another with nearly one-to-one mapping of all the branches. 

> ***Answer 4c***:   As stated above, the K80 model behaves nearly identical to the F84 model. This indicates that the K80 behaves as a special case of the F84 model; as here it can be expected that in the F84 the base frequencies are the same and P(Transition mutation) > P(Transversion mutation) as in the K80 model. Hence, both the models thus behave similarly


### C) ANALYZING A MAXIMUM LIKELIHOOD TREE

In the R code chunk below, do the following:  
1. Read in the maximum likelihood phylogenetic tree used in the handout.
2. Plot bootstrap support values onto the tree

```{r}
ml.bootstrap <- read.tree("./data/ml_tree/RAxML_bipartitions.T1")
par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(ml.bootstrap, type = "phylogram", direction = "right",
show.tip.label = TRUE, use.edge.length = FALSE, cex = 0.6,
label.offset = 1, main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.bootstrap$node.label, font = 2, bg = "white", frame = "r",cex = 0.5)

```

***Question 5***:  

a) How does the maximum likelihood tree compare the to the neighbor-joining tree in the handout? 
If the plots seem to be inconsistent with one another, explain what gives rise to the differences.

b) Why do we bootstrap our tree?

c) What do the bootstrap values tell you? 

d) Which branches have very low support? 

e) Should we trust these branches? 



> ***Answer 5a***:   While the NJ tree has each group nested within a larger group, thus indicating more shared ancestry; thr ML models have two larger groups which further divide into further branches. The differences could be because of the underlying model or because the NJ is not statistically robust.

> ***Answer 5b***:   Repeated resampling of data points results in a probability associated with a particular branch position, and thus increase the weight we assign for each brach of the model. 

> ***Answer 5c***:   The bootstrap values indicate the confidence associated with each branch in the treat. A branch with confidence > 95% can be considered as a "true" branch and not just an artifact.

> ***Answer 5d***:   The ancestral branch for WG42 and WG26; along with the EG42 and LL43F have the lowest support of values 21 and 22 respectively

> ***Answer 5e***:  The branches listed above cannot be trusted as it has a much lower probability of actually being found there in comparison so some of the other branches. 



## 5) INTEGRATING TRAITS AND PHYLOGENY

### A. Loading Trait Database

In the R code chunk below, do the following:  
1. import the raw phosphorus growth data, and  
2. standardize the data for each strain by the sum of growth rates.

```{r}
p.growth <- read.table("./data/p.isolates.raw.growth.txt", sep = "\t", header = TRUE,
row.names = 1)
# Standadize Growth Rates Across Strains
p.growth.std <- p.growth / (apply(p.growth, 1, sum))
```

### B. Trait Manipulations

In the R code chunk below, do the following:  
1. calculate the maximum growth rate ($\mu_{max}$) of each isolate across all phosphorus types,  
2. create a function that calculates niche breadth (*nb*), and  
3. use this function to calculate *nb* for each isolate.

```{r}
# Requires alignment to be read in with as phyDat object
# Calculate Max Growth Rate
umax <- (apply(p.growth, 1, max))

levins <- function(p_xi = ""){
p = 0
for (i in p_xi){
p = p + i^2
}
nb = 1 / (length(p_xi) * p)
return(nb)
}

# Calculate Niche Breadth for Each Isolate
nb <- as.matrix(levins(p.growth.std))
# Add Row & Column Names to Niche Breadth Matrix
rownames(nb) <- row.names(p.growth)
colnames(nb) <- c("NB")
```  

### C. Visualizing Traits on Trees

In the R code chunk below, do the following:  
1. pick your favorite substitution model and make a Neighbor Joining tree,  
2. define your outgroup and root the tree, and  
3. remove the outgroup branch.

```{r}
# Generate Neighbor Joining Tree Using F84 DNA Substitution Model {ape}
nj.tree <- bionj(seq.dist.F84)
# Define the Outgroup
outgroup <- match("Methanosarcina", nj.tree$tip.label)
# Create a Rooted Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)
# Keep Rooted but Drop Outgroup Branch
nj.rooted <- drop.tip(nj.rooted, "Methanosarcina")

```

In the R code chunk below, do the following:  
1. define a color palette (use something other than "YlOrRd"),  
2. map the phosphorus traits onto your phylogeny,  
3. map the *nb* trait on to your phylogeny, and  
4. customize the plots as desired (use `help(table.phylo4d)` to learn about the options).


```{r}
# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))
# Map Phosphorus Traits {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(nj.rooted, p.growth.std)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE,
cex.label = 0.5, scale = FALSE, use.edge.length = FALSE,
edge.color = "black", edge.width = 2, box = FALSE,
col=mypalette(25), pch = 15, cex.symbol = 1.25,
ratio.tree = 0.5, cex.legend = 1.5, center = FALSE)

#Niche Breadth
par(mar=c(1,5,1,5) + 0.1)
x.nb <- phylo4d(nj.rooted, nb)
table.phylo4d(x.nb, treetype = "phylo", symbol = "colors", show.node = TRUE,
cex.label = 0.5, scale = FALSE, use.edge.length = FALSE,
edge.color = "black", edge.width = 2, box = FALSE,
col=mypalette(25), pch = 15, cex.symbol = 1.25, var.label=(" NB"),
ratio.tree = 0.90, cex.legend = 1.5, center = FALSE)
```

***Question 6***:  

a) Make a hypothesis that would support a generalist-specialist trade-off.

b) What kind of patterns would you expect to see from growth rate and niche breadth values that would support this hypothesis?


> ***Answer 6a***:  Generalists would be more likely to occupy a larger niche breadth while being less selective of the type of resource they consume and hence displaying lower growth rates across the resource type. However, specialists would rather key-in on a particular resources and use it to grow faster but which will result in a narrower niche breadth. 

> ***Answer 6b***:  Growth rates would be inversely proportional to niche breadth. If a species is growing extremely well on a resource then that would result in a smaller niche (i.e. narrower niche breadth) in comparison to a species that grows moderately to low on several resources that would result in a larger niche breathe (i.e. broader niche breadth)


## 6) HYPOTHESIS TESTING

### A) Phylogenetic Signal: Pagel's Lambda 

In the R code chunk below, do the following:  
1. create two rescaled phylogenetic trees using lambda values of 0.5 and 0,   
2. plot your original tree and the two scaled trees, and  
3. label and customize the trees as desired.

```{r}
# Visualize Trees With Different Levels of Phylogenetic Signal {geiger}
nj.lambda.5 <- rescale(nj.rooted, "lambda", 0.5)
nj.lambda.0 <- rescale(nj.rooted, "lambda", 0)
layout(matrix(c(1,2,3), 1, 3), width = c(1, 1, 1))
par(mar=c(1,0.5,2,0.5)+0.1)
plot(nj.rooted, main = "lambda = 1", cex = 0.7, adj = 0.5)
plot(nj.lambda.5, main = "lamba = 0.5", cex = 0.7, adj = 0.5)
plot(nj.lambda.0, main = "lamba = 0", cex = 0.7, adj = 0.5)
```

In the R code chunk below, do the following:  
1. use the `fitContinuous()` function to compare your original tree to the transformed trees.

```{r}
fitContinuous(nj.rooted, nb, model = "lambda")
fitContinuous(nj.lambda.5, nb, model = "lambda")
fitContinuous(nj.lambda.0, nb, model = "lambda")
```

***Question 7***:  There are two important outputs from the `fitContinuous()` function that can help you interpret the phylogenetic signal in trait data sets. 
a. Compare the lambda values of the untransformed tree to the transformed (lambda = 0).
b. Compare the Akaike information criterion (AIC) scores of the two models. Which model would you choose based off of AIC score (remember the criteria that the difference in AIC values has to be at least 2)?
c. Does this result suggest that there's phylogenetic signal?

> ***Answer 7a***:  The lambda values for the untransformed tree (lambda = 1) is 0.06 and the values for the transformed tree (lambda = 0) is 0.
#I still have trouble understanding how/why lambda values supplied differ from the lamda values obtained in running this model. 

> ***Answer 7b***: The AIC values for the untransformed tree (lambda = 1) is  -36.913472 and the values for the transformed tree (lambda = 0) is also -37.304586.  The difference in AIC values is ~ 0.7 (not greater than 2) and hence the models can be considered equivalent if not identical

> ***Answer 7c***:  No, it suggests that there is no real phylogenetic signal and the models are equivalent for the transformed and untransformed trees.


### B) Phylogenetic Signal: Blomberg's K 

In the R code chunk below, do the following:  
1. correct tree branch-lengths to fix any zeros,  
2. calculate Blomberg's K for each phosphorus resource using the `phylosignal()` function,  
3. use the Benjamini-Hochberg method to correct for false discovery rate, and  
4. calculate Blomberg's K for niche breadth using the `phylosignal()` function.

```{r}
# First, Correct for Zero Branch-Lengths on Our Tree
nj.rooted$edge.length <- nj.rooted$edge.length + 10^-7
# Calculate Phylogenetic Signal for Growth on All Phosphorus Resources
# First, Create a Blank Output Matrix
p.phylosignal <- matrix(NA, 6, 18)
colnames(p.phylosignal) <- colnames(p.growth.std)
rownames(p.phylosignal) <- c("K", "PIC.var.obs", "PIC.var.mean",
"PIC.var.P", "PIC.var.z", "PIC.P.BH")
# Use a For Loop to Calculate Blomberg's K for Each Resource
for (i in 1:18){
x <- as.matrix(p.growth.std[ ,i, drop = FALSE])
out <- phylosignal(x, nj.rooted)
p.phylosignal[1:5, i] <- round(t(out), 3)
}
# Use the BH Correction on P-values:
p.phylosignal[6, ] <- round(p.adjust(p.phylosignal[4, ], method = "BH"), 3)
p.phylosignal[6, ] 
# Calcualate Phylogenetic Signal for Niche Breadth
signal.nb <- phylosignal(nb, nj.rooted)
signal.nb

p.phylosignal
```

***Question 8***: Using the K-values and associated p-values (i.e., "PIC.var.P"") from the `phylosignal` output, answer the following questions:

a.  Is there significant phylogenetic signal for niche breadth or standardized growth on any of the phosphorus resources?  
b.  If there is significant phylogenetic signal, are the results suggestive of clustering or overdispersion?  

> ***Answer 8a***:   No, the niche breadth does not have a significant phylogenetic signal (p = 0.547) as is true for the phosphorus resources (p >0.05), except for DNA, BGP and cAMP (p < 0.05)
#I'm having trouble understanding why PIC.var.P is chosen as the p-value here and not PIC.P.BH which I assume would be the corrected BH value

> ***Answer 8b***:   Bloomberg's K is < 1; suggestive of overdispersion which means that traits are overdispersed with closely related species less similar than expected by chance.


### C.  Calculate Dispersion of a Trait

In the R code chunk below, do the following:  
1. turn the continuous growth data into categorical data,  
2. add a column to the data with the isolate name,  
3. combine the tree and trait data using the `comparative.data()` function in `caper`, and  
4. use `phylo.d()` to calculate *D* on at least three phosphorus traits.

```{r}
# Turn Continuous Data into Categorical Data
p.growth.pa <- as.data.frame((p.growth > 0.01) * 1)
# Look at Phosphorus Use for Each Resource
apply(p.growth.pa, 2, sum)
# Add Names Column to Data
p.growth.pa$name <- rownames(p.growth.pa)
# Merge Trait and Phylogenetic Data; Run `phylo.d`
p.traits <- comparative.data(nj.rooted, p.growth.pa, "name")
phylo.d(p.traits, binvar = AEP)
phylo.d(p.traits, binvar = DNA)
phylo.d(p.traits, binvar = cAMP)
```

***Question 9***: Using the estimates for *D* and the probabilities of each phylogenetic model, answer the following questions:

a.  Choose three phosphorus growth traits and test whether they are significantly clustered or overdispersed?  
b.  How do these results compare the results from the Blomberg's K analysis?  
c.  Discuss what factors might give rise to differences between the metrics.  

> ***Answer 9a***:The AEP and DNA have D ~> 0.5 (i.e. closer to 1), indicating that they are dispersed such that it is consistent with the expectation of Brownian motion. However, cAMP analysis indicates that D ~ 0, thus making the traits more clumped than random
#I'm concerned about the way the static test is described in the handout versus what is displayed when running the model(phylo.d). The description of p-value testing in the handout appears to indicate a more null hypothesis testing eg. Probability of D being different from 0/ 1. However, the analysis here indicates that the it may be more of a Baeysian model as the text reads"Probability of E(D) resulting from Brownian phylogenetic structure" or "..no (random) phylogenetic structure" where the value of D has a certain % chance of being explained due to Brownian phylo or random phylo.

> ***Answer 9b***:  The Bloomberg K shows that all species have K = 0; indicating that all the traits are over-dispersed. 

> ***Answer 9c***:  While Blomberg looks at how the trait distributions on trees would evolve under random motion, the latter looks at how the trait is distributed across the phylogenetic trees space.  

## 7) PHYLOGENETIC REGRESSION

In the R code chunk below, do the following:  
1. Load and clean the mammal phylogeny and trait dataset,
2. Fit a linear model to the trait dataset, examining the relationship between mass and BMR,
2. Fit a phylogenetic regression to the trait dataset, taking into account the mammal supertree

```{r}
# Input the tree and dataset
mammal.Tree <- read.tree("./data/mammal_best_super_tree_fritz2009.tre")
mammal.data <- read.table("./data/mammal_BMR.txt", sep = "\t",
header = TRUE)
# Select the variables we want to analyze
mammal.data <- mammal.data[, c("Species", "BMR_.mlO2.hour.",
"Body_mass_for_BMR_.gr.")]
mammal.species <- array(mammal.data$Species)
# Select the tips in the mammal tree that are also in the
# dataset
pruned.mammal.tree <- drop.tip(mammal.Tree, mammal.Tree$tip.label[-na.omit(match(mammal.species,
mammal.Tree$tip.label))])
# Select the species from the dataset that are in our prunned
# tree
pruned.mammal.data <- mammal.data[mammal.data$Species %in% pruned.mammal.tree$tip.label,
]
# Turn column of Species names into rownames
rownames(pruned.mammal.data) <- pruned.mammal.data$Species
fit <- lm(log10(BMR_.mlO2.hour.) ~ log10(Body_mass_for_BMR_.gr.),
data = pruned.mammal.data)
plot(log10(pruned.mammal.data$Body_mass_for_BMR_.gr.), log10(pruned.mammal.data$BMR_.mlO2.hour.),
las = 1, xlab = "Body mass (kg), log", ylab = "Basal Metabolic Rate (BMR), log")
abline(a = fit$coefficients[1], b = fit$coefficients[2])
b1 <- round(fit$coefficients[2], 3)
eqn <- bquote(italic(z) == .(b1))
# plot the slope
text(0.5, 4.5, eqn, pos = 4)

#install.packages("phylolm")
require("phylolm")
# Run a phylogeny-corrected regression with no bootstrap
# replicates
fit.phy <- phylolm(log10(BMR_.mlO2.hour.) ~ log10(Body_mass_for_BMR_.gr.), data = pruned.mammal.data, pruned.mammal.tree, model = "lambda", boot = 0)

plot(log10(pruned.mammal.data$Body_mass_for_BMR_.gr.), log10(pruned.mammal.data$BMR_.mlO2.hour.),
las = 1, xlab = "Body mass (kg), log", ylab = "Basal Metabolic Rate (BMR), log")
abline(a = fit.phy$coefficients[1], b = fit.phy$coefficients[2])
b1.phy <- round(fit.phy$coefficients[2], 3)
eqn <- bquote(italic(z) == .(b1.phy))
text(0.5, 4.5, eqn, pos = 4)

AIC(fit, fit.phy)
```
  

a. Why do we need to correct for shared evolutionary history?
b. How does a phylogenetic regression differ from a standard linear regression?
c. Interpret the slope and fit of each model. Did accounting for shared evolutionary history improve or worsten the fit?
d. Try to come up with a scenario where the relationship between two variables would completely disappear when the underlying phylogeny is accounted for.

> ***Answer 10a***:  Correcting for shared evolutionary history inherently circumvents the problem of assuming that all points are independent, when infact that they may have a certain degree of dependence on each other depending on their phylogenetic structure. 

> ***Answer 10b***:  While standard linear regression assumes a mean 0 and normally distributed variance; phylogenetic regressions assumes that the variance assumes a covariance matrix which in turn accounts for branch length between species.

> ***Answer 10c***:  While the linear regression indicates that a unit change in log(BMR) would result in 0.6 times the unit change in body mass; phylogenetic regression indicates that  a unit change in log(BMR) would result in 0.712 times the unit change in body mass. Accounting for phylogeny strengthened the fit of the model, by AIC values (fit.phy = -646.9, fit = -365.5). 
#Unsure if AIC values need to be taken as absolute values in this case, or if it should be considered with the negative sign which would change the inference. I tried to compare the R2 of fit with fit.phy; however, fit.phy produces no R2 and it is not clear how one can compare AIC to R2 value

> ***Answer 10d***:  The relationship between two variables could completely disappear if any variation was explained by phylogeny alone and not a result of any other external factors that might explain variation in their relationship.
#Dabbling with how the tree for this type of relationship will look like


## 7) SYNTHESIS

Below is the output of a multiple regression model depicting the relationship between the maximum growth rate ($\mu_{max}$) of each bacterial isolate and the niche breadth of that isolate on the 18 different sources of phosphorus. 
One feature of the study which we did not take into account in the handout is that the isolates came from two different lakes. 
One of the lakes is an very oligotrophic (i.e., low phosphorus) ecosystem named Little Long (LL) Lake. 
The other lake is an extremely eutrophic (i.e., high phosphorus) ecosystem named Wintergreen (WG) Lake.
We included a "dummy variable" (D) in the multiple regression model (0 = WG, 1 = LL) to account for the environment from which the bacteria were obtained. For the last part of the assignment, plot nich breadth vs. $\mu_{max}$ and the slope of the regression for each lake. Be sure to color the data from each lake differently. 

```{r, echo=FALSE, fig.width=6, fig.height=4}
p.growth <- read.table("./data/p.isolates.raw.growth.txt", sep = "\t", header = TRUE, row.names = 1)
umax <- (apply(p.growth, 1, max)) # calculate max growth
lake <- ifelse(grepl("WG",row.names(p.growth)),'WG', 'LL') # make an empty vector for lake id
tradeoff <- data.frame(nb,umax,lake) # make new data frame

D <- (lake == "LL") * 1
fit.lakeLL <-lm(log10(umax) ~ nb + D + nb * D)

plot(nb + D + nb * D, log10(umax), las = 1, ylab = "Max. Growth Rate, log10", xlab = "Niche Breadth", col = "red", main = "Relationship for Low Phosporus Lake (LL)")
abline(a = fit.lakeLL$coefficients[1], b = fit.lakeLL$coefficients[2])
b1.lakeLL <- round(fit.lakeLL$coefficients[2], 3)
eqn <- bquote(italic(z) == .(b1.lakeLL))
text(0.3, -0.5, eqn, pos = 4)
title()

D <- (lake == "WG") * 1
fit.lakeWG <-lm(log10(umax) ~ nb + D + nb * D)

plot(nb + D + nb * D, log10(umax), las = 1, ylab = "Max. Growth Rate, log10", xlab = "Niche Breadth", col = "green",main = "Relationship for High Phosporus Lake (WG)")
abline(a = fit.lakeWG$coefficients[1], b = fit.lakeWG$coefficients[2])
b1.lakeWG <- round(fit.lakeWG$coefficients[2], 3)
eqn <- bquote(italic(z) == .(b1.lakeWG))
text(0.3, -0.5, eqn, pos = 4)
```

***Question 11***: Based on your knowledge of the traits and their phylogenetic distributions, what conclusions would you draw about our data and the evidence for a generalist-specialist tradeoff? 

> ***Answer 11***: It appears that there is evidence for generalist-specialist trade-off here. As hypothesized earlier, increasing niche breadth results in decrease in maximum growth rate i.e. more a species expands, less likely that it is able to consume ALL resources with equal voraciousness when compared to a specialist. 

